src/engine/Account-Imap.cpp

```diff
@@ -0,0 +1,1186 @@
+
+    std::string str_key (key);
+
+    if (!category)
+        kvp_path = IMAP_FRAME + delim + str_key;
+    else
+    {
+        std::string str_category (category);
+        kvp_path = IMAP_FRAME + delim + str_category + delim + str_key;
+    }
+    xaccAccountBeginEdit (account);
+
+    if (qof_instance_has_slot (QOF_INSTANCE (account), kvp_path.c_str()))
+    {
+        qof_instance_slot_delete (QOF_INSTANCE (account), kvp_path.c_str());
+
+        if (category)
+        {
+            std::string str_category (category);
+            kvp_path = IMAP_FRAME + delim + str_category;
+
+            qof_instance_slot_delete_if_empty (QOF_INSTANCE (account), kvp_path.c_str());
+        }
+        qof_instance_slot_delete_if_empty (QOF_INSTANCE (account), IMAP_FRAME);
+    }
+    qof_instance_set_dirty (QOF_INSTANCE (account));
+    xaccAccountCommitEdit (account);
+}
+
+Account *
+GncImportMatchMap::find_account_bayes (GList* tokens)
+{
+std::cout << "find_account_bayes" << std::endl;
+    return nullptr;
+}
+
+void
+GncImportMatchMap::add_account_bayes (GList* tokens, Account *acc)
+{
+std::cout << "add_account_bayes" << std::endl;
+}
+
+/*****************************************************************************/
+
+// C access functions
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+GncImportMatchMap *
+gnc_account_imap_create_imap (Account *acc)
+{
+std::cout << "c_create_imap" << std::endl;
+    if (!acc) return nullptr;
+
+    GncImportMatchMap *ImportMap = new GncImportMatchMap(acc);
+    return ImportMap;
+}
+
+void
+gnc_account_imap_delete_imap (GncImportMatchMap *imap)
+{
+std::cout << "c_delete_imap" << std::endl;
+    if (!imap) return;
+    delete (imap);
+}
+
+Account *
+gnc_account_imap_find_account (GncImportMatchMap *imap, const char* category, const char *key)
+{
+std::cout << "c_find_account" << std::endl;
+
+    return imap->find_account (category, key);
+}
+
+void
+gnc_account_imap_add_account (GncImportMatchMap *imap, const char* category, const char *key, Account *acc)
+{
+std::cout << "c_add_account" << std::endl;
+
+    imap->add_account (category, key, acc);
+}
+
+void
+gnc_account_imap_delete_account (GncImportMatchMap *imap, const char *category, const char *key)
+{
+std::cout << "c_delete_account" << std::endl;
+
+    imap->delete_account (category, key);
+}
+
+Account *
+gnc_account_imap_find_account_bayes (GncImportMatchMap *imap, GList* tokens)
+{
+std::cout << "c_find_account_bayes" << std::endl;
+
+    return imap->find_account_bayes (tokens);
+}
+
+void gnc_account_imap_add_account_bayes (GncImportMatchMap *imap, GList* tokens, Account *acc)
+{
+std::cout << "c_add_account_bayes" << std::endl;
+
+    imap->add_account_bayes (tokens, acc);
+}
+
+#ifdef __cplusplus
+} /* closing brace for extern "C" */
+#endif
+
+
+
+
+/*****************************************************************************/
+
+#ifdef skip
+/* Obtain an ImportMatchMap object from an Account or a Book */
```

Comment:
This would be clearer if it was #ifndef __cplusplus, but better still just remove the old C code.